src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeView.cs

```diff
@@ -0,0 +1,381 @@
+			}
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
+
+			if (property == SelectionModeProperty && ListControl != null)
+			{
+				TreeViewSelectionMode value = SelectionMode;
+				switch (value)
+				{
+					case TreeViewSelectionMode.None:
+						{
+							ListControl.SelectionMode = ListViewSelectionMode.None;
+							UpdateItemsSelectionMode(false);
+						}
+						break;
+
+					case TreeViewSelectionMode.Single:
+						{
+							ListControl.SelectionMode = ListViewSelectionMode.Single;
+							UpdateItemsSelectionMode(false);
+						}
+						break;
+
+					case TreeViewSelectionMode.Multiple:
+						{
+							ListControl.SelectionMode = ListViewSelectionMode.None;
+							UpdateItemsSelectionMode(true);
+						}
+						break;
+				}
+			}
+			else if (property == ItemsSourceProperty)
+			{
+				m_rootNode.IsContentMode = true;
+
+				if (ListControl != null)
+				{
+					var viewModel = ListControl.ListViewModel;
+					viewModel.IsContentMode = true;
+				}
+
+				m_rootNode.ItemsSource = ItemsSource;
+			}
+			else if (property == SelectedItemProperty)
+			{
+				var items = SelectedItems;
+				var selected = items.Count > 0 ? items[0] : null;
+				if (args.NewValue != selected)
+				{
+					ListControl?.ListViewModel?.SelectSingleItem(args.NewValue);
+				}
+			}
+		}
+
+		private void OnListControlDragItemsStarting(object sender, DragItemsStartingEventArgs args)
+		{
+			var treeViewArgs = new TreeViewDragItemsStartingEventArgs(args);
+			DragItemsStarting?.Invoke(this, treeViewArgs);
+		}
+
+		private void OnListControlDragItemsCompleted(object sender, DragItemsCompletedEventArgs args)
+		{
+			object CreateNewParent(IReadOnlyList<object> items, TreeViewList listControl, TreeViewNode rootNode)
+			{
+				if (listControl != null && items != null && items.Count > 0)
+				{
+					var draggedNode = listControl.NodeFromItem(items[0]);
+					if (draggedNode != null)
+					{
+						var parentNode = draggedNode.Parent;
+						if (parentNode != null && parentNode != rootNode)
+						{
+							return ListControl.ItemFromNode(parentNode);
+						}
+					}
+				}
+				return null;
+			}
+			var newParent = CreateNewParent(args.Items, ListControl, m_rootNode);
+
+			var treeViewArgs = new TreeViewDragItemsCompletedEventArgs(args, newParent);
+			DragItemsCompleted?.Invoke(this, treeViewArgs);
+		}
+
+		private void OnListControlSelectionChanged(object sender, SelectionChangedEventArgs args)
+		{
+			if (SelectionMode == TreeViewSelectionMode.Single)
+			{
+				RaiseSelectionChanged(args.AddedItems, args.RemovedItems);
+
+				object GetNewSelectedItem(SelectionChangedEventArgs args)
+				{
+					var newItems = args.AddedItems;
+					if (newItems != null)
+					{
+						if (newItems.Count > 0)
+						{
+							return newItems[0];
+						}
+						else
+						{
+							return null;
+						}
+					}
+					else
+					{
+						return null;
+					}
+				};
```

Comment:
Codacy found an issue: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)